app/controllers/discovered_hosts_controller.rb

```diff
@@ -84,13 +84,27 @@ def update
   end
 
   def refresh_facts
-    if @host.is_a?(::Host::Discovered) and @host.refresh_facts
+    if @host.is_a?(::Host::Discovered) && @host.refresh_facts
       process_success :success_msg => _("Facts refreshed for %s") % @host.name, :success_redirect => :back
     else
       process_error :error_msg => _("Failed to refresh facts for %s") % @host.name, :redirect => :back
     end
   end
 
+  def reboot
+    unless @host.is_a?(::Host::Discovered)
+      process_error :error_msg => _("Host of type %s can not be rebooted") % @host.type, :redirect => :back
+    end
+
+    if @host.reboot
+      process_success :success_msg => _("Rebooting host %s") % @host.name, :success_redirect => :back
+    else
+      process_error :error_msg => _("Failed to reboot host %s") % @host.name, :redirect => :back
+    end
+    rescue  => e
+      process_error :error_msg => _("Failed to reboot host %s with error %s") % [@host.name, e.message], :redirect => :back
```

Comment:
Use hash of parameters instead of interpolation (e.g., ```_("Failed to reboot host %{hostname} with error %{message}") % {:hostname => @host.name, :message => e.message} ```